python/lsst/meas/base/diaCalculationPlugins.py

```diff
@@ -79,6 +86,343 @@ def wrapperCatchWarnings(*args, **kwargs):
+                                      np.isnan(df["midpointMjdTai"]))]
+
+            if (len(tmpDf)) < min_detections:
+                return pd.Series({periodCol: np.nan,
+                                  powerCol: np.nan,
+                                  fapCol: np.nan,
+                                  ampCol: np.nan,
+                                  phaseCol: np.nan})
+
+            time = tmpDf["midpointMjdTai"].to_numpy()
+            flux = tmpDf["psfFlux"].to_numpy()
+            flux_err = tmpDf['psfFluxErr'].to_numpy()
+            bands = tmpDf['band'].to_numpy()
+
+            # TODO: currently assuming 1-1 base-band model.
+            lsp = lspm(time, flux, bands, dy=flux_err,
+                       nterms_base=1, nterms_band=1)
+
+            f_grid = compute_optimized_periodogram_grid(
+                time, oversampling_factor=oversampling_factor, nyquist_factor=nyquist_factor)
+            period = 1/f_grid
+            power = lsp.power(f_grid)  # compute power
+
+            # Calculate the False-Alarm probability Baluev approximation
+            fap_estimate = calculate_baluev_fap(
+                time, len(time), period[np.argmax(power)], np.max(power))
+
+            def generate_lsp_params(lsp_model, fbest, bands):
+                """Generate the Lomb-Scargle parameters.
+
+                    Parameters:
+                    ----------
+                    lsp_model : `astropy.timeseries.LombScargleMultiband`
+                        The Lomb-Scargle model.
+                    fbest : `float`
+                        The best period.
+                    bands : `array`
+                        The bands of the time series.
+
+                    Returns:
+                    -------
+                    Amp : `array`
+                        The amplitude of the time series.
+                    Ph : `array`
+                        The phase of the time series.
+
+                    References:
+                    ----------
+                    .. [1] VanderPlas, J. T., & Ivezic, Z. 2015, ApJ, 812, 18
+                """
+                # Calculate based on the best period the parameters
+                best_params = lsp_model.model_parameters(fbest, units=True)
+
+                name_params = [f'theta_base_{i}' for i in range(3)]
+                name_params += [f'theta_band_{band}_{index}' for band in np.unique(
+                    bands) for index in range(3)]
+
+                df_params = pd.DataFrame([best_params], columns=name_params)
+
+                # compute amplitude and phase
+                Amplitude_band = [np.sqrt(df_params[f'theta_band_{band}_1']**2
+                                          + df_params[f'theta_band_{band}_2']**2)
+                                  for band in np.unique(bands)]
+                Phase_bands = [np.arctan2(df_params[f'theta_band_{band}_2'],
+                                          df_params[f'theta_band_{band}_1']) for band in np.unique(bands)]
```

Comment:
If you're reusing the result of `np.unique`, you should assign it to a variable. That's a non-trivial calculation, so it's worth only doing once.